core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -529,6 +531,7 @@ public TableMetadata replaceProperties(Map<String, String> rawProperties) {
 
     return new Builder(this)
         .setProperties(updated)
+        .setRef(SnapshotRef.MAIN_BRANCH, SnapshotRef.branchBuilder(currentSnapshotId).build())
```

Comment:
I don't think it is a good idea to overwrite the main branch anywhere. If there are no refs, but there is a current snapshot, then the builder should produce a main branch. I think that's the only place where this logic should be handled.

Reply:
Good point, I should be setting this in the builder as that's the whole point of it. I'll update this